packages/js/src/session/session.ts

```diff
@@ -88,4 +88,9 @@ export class Session {
       this.#emitter.emit('session.initialize.resolved', { args: this.#options, error });
     }
   }
+
+  public async reinitialize(newOptions: InitializeSessionArgs): Promise<void> {
+    this.#options = newOptions;
+    await this.initialize();
+  }
```

Comment:
As a general note,  I wonder if a new method is needed, and not just calling the re-initialize from the provider itself

Reply:
i thought about it as well, but then we would need to expose a setter for this.#options instead.